### ISSUE TITLE
Build New Mongo -> MSSQL Data Transformation Service

### ISSUE THREAD

**Issue by shanemelliott** [department-of-veterans-affairs/va.gov-team#20837]
Currently VEText has a manual node.js data transformation service that moves data from mongo (operational Database) to MSSQL (Reporting Database).  This service needs to be re-written and scheduled to be more reliable and add individual record updating. 

Tables/collections

- 'App.MON_ConvertedStats'/ 'convertedAppointmentsStats'
- 'DFlt.MON_COVIDAppts'/'covidVaccineAppointments'
- 'Dflt.MON_Login'/ 'portalLogin'
- 'Dflt.MON_SiteStats'/ 'outboundAppointmentRemindersSummary'
- 'DFlt.MON_SurgTracking'/ 'surgeryPatients'
- 'Dflt.MON_PatTracking'/ 'patientTracking'
- 'MON_COVIDAppts'/:'covidVaccineAppointments'

All of the fields are not transferred, a configuration process that identifies the fields for each table to be transferred.
Dates need to be transformed from Fileman to date/time and timeZone needs to be adjusted for some of the dates. 
needs to be scheduled daily. 
options:

- java
- node.js
- etil

shane went to bulk upload because otherwise it was failing on inserts. longterm plan was to use bulk upload - that's why there are two directories. the only one bulk upload was done for was covid 19. there is batch uploading but not bulk uploading in java

1. Set up an environment with correct test data for mongoDB and sql server.  outline any funky data that comes through. (s)
2. Take three steps separately (data grabbed from mongo - (translation) - insert into sql)
3. Move it into va, test in that environment

right now you need elevated priv to get onto any va servers - we will build it then we have a plan for how we will maintain and deploy when we get there.

**Comment by shanemelliott** [comment]
[https://app.zenhub.com/files/133843125/f281b06f-96b2-4357-a5f7-c2bcbc266cba/download](https://app.zenhub.com/files/133843125/f281b06f-96b2-4357-a5f7-c2bcbc266cba/download)

**Comment by shanemelliott** [comment]
[https://app.zenhub.com/files/133843125/b62748a3-452c-4d72-9638-037d4d3b1af9/download](https://app.zenhub.com/files/133843125/b62748a3-452c-4d72-9638-037d4d3b1af9/download)

Added Sql Table create scripts.

**Comment by shanemelliott** [comment]
Sent Kevin JSON Docs for the Mongo collections.

**Comment by kjduensing** [comment]
tl;dr: Decision: Consider Fileman timestamps to be "timezone-less" for now.

Regarding UTC vs. local server time: The Fileman timestamps in the MongoDB database are inserted as the local time of the server (wherever it is). Shane has been doing modifications of these timestamps once the data is in the SQL database based on a list of station numbers and timezones. 

After talking to Shane, we decided to consider the Fileman timestamps as "wall-clock" time with no timezone offset. This simplifies the code a bit as we would have to consider the timezone of the VEText Server as well as the VistA server for certain fields.

**Comment by kjduensing** [comment]
Progress: Docker images spun up locally

**Comment by kjduensing** [comment]
We talked a little about this in the 3/10 VEText mtg. In lieu of Morphia, I am using the bson library that is packaged with the mongodb-client "uberjar": https://mvnrepository.com/artifact/org.mongodb/mongo-java-driver

I referenced this article from the mongo developer site: https://developer.mongodb.com/quickstart/java-mapping-pojos/?utm_campaign=javapojos

**Comment by Kennon-git** [comment]
Wednesday 17 Mar 2021 - Covid table test

**Comment by kjduensing** [comment]
Blocked - need PIV card && GFE to test in real environment
